src/types.rs

```diff
@@ -494,20 +511,21 @@ pub fn processed_to_device_event_to_js_value(
 ) -> JsValue {
     match processed_to_device_event {
         matrix_sdk_crypto::types::ProcessedToDeviceEvent::Decrypted { raw, encryption_info } => {
-            JsValue::from(DecryptedToDeviceEvent {
-                processed_type: ProcessedToDeviceEventType::Decrypted,
-                wire_event: raw.json().get().to_owned().into(),
-                encryption_info: encryption_info.into(),
-            })
+            JsValue::from(DecryptedToDeviceEvent::new(
+                raw.json().get().into(),
+                encryption_info
+                    .try_into()
+                    .expect("to-device encryption info will always convert to OlmEncryptionInfo"),
```

Comment:
it might be good to explain in more detail why this is the case.

I think it might be better just to drop such events. (Easiest way to do that: have `processed_to_device_event_to_js_value` return `Option<JsValue>` and use `filter_map` instead of `map` at the call site.)